#!/usr/bin/env python
PACKAGE = "masterslave"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("rosRate", int_t, 0, "ros rate", 1000, 20, 1000)

state_enum = 	gen.enum([gen.const("IDLE", int_t, 0, "Idle Mode"),
			  gen.const("FREE", int_t, 1, "Grav Comp"),
			  gen.const("MOVE_TO_POSE", int_t, 2, "Masterslave first attempt")],
			"An enum to set size")

gen.add("cur_state", int_t, 0, "A size parameter which is edited via an enum", 0, 0, 2, edit_method=state_enum)
gen.add("set_Trocar", bool_t, 0, "Set the trocar", False)
gen.add("isReality", bool_t, 0, "Is the tool used in reality or in simulation", False)

exit(gen.generate(PACKAGE, "MasterSlave", "MasterSlave"))
